### ISSUE TITLE
Route Decision | Route Claim | Partial Grants & Remands with no EP | Change Location, Add Diary note

### ISSUE THREAD

**Issue by lakohl** [department-of-veterans-affairs/caseflow#884]
As a Caseworker, I would like to have a message informs me of how the appeal will be routed to the appropriate place via VACOLS when I am working on a Partial Grant or Remand case that has a special issue for which we cannot create an EP, so that I am aware of next steps.

# Acceptance Criteria
1.  Verify that after clicking the **_Route Claim_** button on the `Decision Review` page for a Partial Grant or Remand that an EP cannot be created, the `Route Claim - VACOLS` page is displayed with the following Title, Informational Alert message, and subtext:  (Refer to Mockup)  _Note:  This is done instead of forcing them to cancel out._

     **Title**

     > Route Claim

     **Informational Message**

     > We are unable to create an EP for claims with this Special Issue.

     **Subtext**

     > To ensure this claim is routed correctly, we will take the following steps in VACOLS:
1. Verify that on the `Route Claim - VACOLS` page, the the following data elements are displayed (refer to mockup):

     - **Change location to** - New location code that will be used to update VACOLS Location field.  _Display Only._
          _Format_:  [Location_Code]
     - **Add the diary note** - pre-filled diary note verbiage that will be used to update VACOLS Diary Note field.  _Ediitable_.

1. Verify that the "Change location to" field displays the [Location_Code] as indicated for the following Special Issue:

     | ID | Special Issue | New VACOLS Location |
     | --- | --- | --- |
     1|Education Debt |50
     2|Waiver of Overpayment |50
     3|Insurance |50
     4|Foreign Claim |50
     5|Pension |50
     6|DIC, death, accrued benefits |50
     7|VAMC |51
     8|National Cemetery Administration |53
1. Verify that the _"Add diary note"_ field is pre-populated with the following verbiage: 

     > The BVA [Decision_Type] decision dated [Decision_Date] for [Veteran_FirstName] [Veteran_LastName], ID [Veteran_ID], was sent to the ARC but cannot be processed here, as it contains [Special_Issue_Category] in your jurisdiction. Please proceed with control and implementation of this grant.

1. Verify that the following links are available: 
     - **_Back to Decision Review_** 
     - **_Cancel_**
1. Verify that the following button is available:  **_Finish Routing Claim_**
1. Verify that after clicking the **_Finish Routing Claim_** button, the `Confirmation Congratulation` page is displayed with the following checklist items:

     - Decision Reviewed
     - Routed Claim in VACOLS
1. Verify that after clicking on the **_Cancel_** link, the `Stop Processing Claim` modal is displayed.
1. Verify that after clicking on the **_Back to Decision Review_** link, the `Caseworker Dashboard` page is displayed.
1. Verify that if all the text in the diary note field is removed and the caseworker clicks enter, we will throw an error indicating that the notes field is required.
1. Verify that clicking the **_Finish Routing Claim_** button, the VACOLS Location field is updated with the "Change location to" value.

1. Verify that after clicking the **_Finish Routing Claim_** button, the VACOLS Dairy Note field is updated with the "Add the diary note" value.

## Mockups
Route claim page
<img width="699" alt="screen shot 2017-02-13 at 9 35 11 am" src="https://cloud.githubusercontent.com/assets/4975959/22887347/b8066a78-f1cf-11e6-9456-8824406830f7.png">

Congratulations page
<img width="650" alt="screen shot 2017-02-09 at 7 52 56 pm" src="https://cloud.githubusercontent.com/assets/4975959/22809567/62a583b4-ef01-11e6-8955-3fe7bfb371e1.png">

## Notes
- What happens for EP's that aren't routed to ARC, but do have a special issue? (@lakohl)
- When this story is completed, this issue will be unblocked for validation (https://github.com/department-of-veterans-affairs/caseflow/issues/723)
- We are not changing the ROJ at this time. Please don't include it on the page

# Related Stories
[Email Addresses - Routing Claims](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Dispatch/Routing%20EPs/Email%20Routing%20Table.md)

**Comment by lakohl** [comment]
@shanear adding you for awareness

**Comment by lakohl** [comment]
This is very similar to https://github.com/department-of-veterans-affairs/caseflow/issues/720 shall we combine them?

**Comment by shanear** [comment]
@lakohl Yes. They look the exact same. Also, can we take out the ROJ routing into another issue?

**Comment by mdbenjam** [comment]
@abbyraskin  Do we have a list of ROJ codes? I used this list for locations from #720:

Education Debt - Location 50
Waiver of Overpayment - Location 50
Insurance - Location 50
Foreign Claim - Location 50
Pension - Location 50
DIC, death, accrued benefits - Location 50
VAMC - Location 51
National Cemetery Administration - Location 53

**Comment by abbyraskin** [comment]
@mdbenjam are you interested in the RO field in VACOLS or the location field? anything that gets routed to the ROJ goes to location 50.

**Comment by abbyraskin** [comment]
@mdbenjam are you interested in the list of RO codes in VACOLS or the location codes? anything that gets routed to the ROJ goes to location 50.

**Comment by mdbenjam** [comment]
@abbyraskin I'm interested in both. Do I just base these values off what Sharon used in her ticket for emailing offices for the full grant case? The information in these lists? https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Dispatch/Routing%20EPs/Email%20Routing%20Table.md

**Comment by mdbenjam** [comment]
Waiting on the resolution to this: https://github.com/department-of-veterans-affairs/caseflow/issues/1043 before proceeding.

**Comment by kierachell** [comment]
Same issue as #723:
```
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] Started POST "/dispatch/establish-claim/1005/review-complete" for 127.0.0.1 at 2017-03-09 09:07:57 -0500
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] Processing by EstablishClaimsController#review_complete as JSON
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   Parameters: {"vacols_note"=>"The BVA Full Grant decision dated 02/27/2017 is being transfered from ARC as it contains: Education - GI Bill, dependents educational assistance, scholarship, transfer of entitlement in your jurisdiction.", "task_type"=>:EstablishClaim, "id"=>"1005", "establish_claim"=>{}}
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36mUser Load (1.2ms)[0m  [1mSELECT  "users".* FROM "users" WHERE "users"."css_id" = $1 AND "users"."station_id" = $2 LIMIT 1[0m  [["css_id", "CASEFLOW1"], ["station_id", "283"]]
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[35m (1.2ms)[0m  BEGIN
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36m (1.1ms)[0m  [1mCOMMIT[0m
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[35m (1.0ms)[0m  BEGIN
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36mTask Load (1.3ms)[0m  [1mSELECT  "tasks".* FROM "tasks" WHERE "tasks"."id" = $1 LIMIT 1[0m  [["id", 1005]]
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[35mAppeal Load (1.4ms)[0m  SELECT  "appeals".* FROM "appeals" WHERE "appeals"."id" = $1 LIMIT 1  [["id", 140]]
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36m (1.4ms)[0m  [1mROLLBACK[0m
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] Completed 500 Internal Server Error in 215ms (ActiveRecord: 8.7ms)
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] ** [Raven] Sending event 5dbf5719a3c2872b0968ea09e19b0259 to Sentry
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] 
ArgumentError (unknown keyword: station):
  app/services/appeal_repository.rb:201:in `location_after_dispatch'
  app/services/appeal_repository.rb:194:in `update_location_after_dispatch!'
  app/services/appeal_repository.rb:183:in `block in update_vacols_after_dispatch!'
  app/services/appeal_repository.rb:182:in `update_vacols_after_dispatch!'
  app/services/dispatch.rb:73:in `update_vacols!'
  app/controllers/establish_claims_controller.rb:20:in `block in review_complete'
  app/controllers/establish_claims_controller.rb:19:in `review_complete'
  app/middleware/metrics_auth.rb:10:in `call'
```
UI is lame, though: @lakohl I see no clear mocks for this page, please update:
<img width="1012" alt="screen shot 2017-03-09 at 9 07 52 am" src="https://cloud.githubusercontent.com/assets/18075411/23753580/073eb498-04a8-11e7-895a-234a76acc60e.png">

**Comment by kierachell** [comment]
Same issue as #723:
## Cloudwatch Log:
```
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] Started POST "/dispatch/establish-claim/1005/review-complete" for 127.0.0.1 at 2017-03-09 09:07:57 -0500
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] Processing by EstablishClaimsController#review_complete as JSON
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   Parameters: {"vacols_note"=>"The BVA Full Grant decision dated 02/27/2017 is being transfered from ARC as it contains: Education - GI Bill, dependents educational assistance, scholarship, transfer of entitlement in your jurisdiction.", "task_type"=>:EstablishClaim, "id"=>"1005", "establish_claim"=>{}}
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36mUser Load (1.2ms)[0m  [1mSELECT  "users".* FROM "users" WHERE "users"."css_id" = $1 AND "users"."station_id" = $2 LIMIT 1[0m  [["css_id", "CASEFLOW1"], ["station_id", "283"]]
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[35m (1.2ms)[0m  BEGIN
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36m (1.1ms)[0m  [1mCOMMIT[0m
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[35m (1.0ms)[0m  BEGIN
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36mTask Load (1.3ms)[0m  [1mSELECT  "tasks".* FROM "tasks" WHERE "tasks"."id" = $1 LIMIT 1[0m  [["id", 1005]]
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[35mAppeal Load (1.4ms)[0m  SELECT  "appeals".* FROM "appeals" WHERE "appeals"."id" = $1 LIMIT 1  [["id", 140]]
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ]   [1m[36m (1.4ms)[0m  [1mROLLBACK[0m
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] Completed 500 Internal Server Error in 215ms (ActiveRecord: 8.7ms)
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] ** [Raven] Sending event 5dbf5719a3c2872b0968ea09e19b0259 to Sentry
[certification.cf.uat.ds.va.gov] [172.30.70.123] [2017-03-09 09:07:57 -0500] [CASEFLOW1 ] 
ArgumentError (unknown keyword: station):
  app/services/appeal_repository.rb:201:in `location_after_dispatch'
  app/services/appeal_repository.rb:194:in `update_location_after_dispatch!'
  app/services/appeal_repository.rb:183:in `block in update_vacols_after_dispatch!'
  app/services/appeal_repository.rb:182:in `update_vacols_after_dispatch!'
  app/services/dispatch.rb:73:in `update_vacols!'
  app/controllers/establish_claims_controller.rb:20:in `block in review_complete'
  app/controllers/establish_claims_controller.rb:19:in `review_complete'
  app/middleware/metrics_auth.rb:10:in `call'
```
## Sentry: https://10.236.66.247/sentry/caseflow-uat/issues/114/
UI is lame, though: @lakohl I see no clear mocks for this page, please update:
<img width="1012" alt="screen shot 2017-03-09 at 9 07 52 am" src="https://cloud.githubusercontent.com/assets/18075411/23753580/073eb498-04a8-11e7-895a-234a76acc60e.png">

**Comment by kierachell** [comment]
**PASSED**

validated on `uat`
![partialnote](https://cloud.githubusercontent.com/assets/18075411/23761777/eef92c46-04c1-11e7-9886-1ec60d37495d.gif)
![remandnote](https://cloud.githubusercontent.com/assets/18075411/23761778/ef2a6996-04c1-11e7-91b3-ded622ea248d.gif)


all specified SIC work with Note and Location updated in VACOLS
